### ISSUE TITLE
QwCorrelator needs ErrorFlag cut (demo on run 2962)

### ISSUE THREAD

**Issue by wdconinc** [JeffersonLab/japan#206]
### Environment: (where does this bug occur, have you tried other environments)
- branch (master for latest released): develop
- revision (HEAD for most recent): HEAD
- OS or system: ifarm
- Special ROOT or Geant4 versions? 6.16

### Steps to reproduce: (give a step by step account of how to trigger the bug)
1. Disable sam7 (too many errors during this run) in prex_corrolator.conf, disable dv cav*, add iv diff_bcm12X
2. Make similar changes to prex_corrolator_all.conf
3. build/qwparity -c prex.conf -r 2962 -e :100k

### Expected Result: (what do you expect when you execute the steps above)
Correlations should behave statistically and not be affected by outliers, in particular RMS width of initial distributions should be identical modulo a few dropped events due to more ivs in the _all set.

### Actual Result: (what do you get when you execute the steps above)
The widths (i.e. errors on uncorrected means printed by correlator at end of run) are very different (50% larger for the prex_corrolator set, compared to the more restrictive data quality in prex_corrolator_all.conf, even though number of patterns is 24905 vs 24888 only different by 17 patterns).

Looking at this in more detail, it turns out the correlator does not apply the ErrorFlag cut:
Top row: bcm, left: without ErrorFlag == 0 cut, right: with ErrorFlag cut
Bottom row: asym_sam1 with all Device_Error_Code == 0 cuts on ivs and dvs in prex_corrolator.conf (i.e. 24905 patterns), left: without ErrorFlag == 0 cut, right: with ErrorFlag cut 
![image](https://user-images.githubusercontent.com/4656391/60520913-37756e80-9cb4-11e9-872f-e6a7b8c38a64.png)

The correlations we are interested in are those after ErrorFlag cut, not before. Other data handlers (alarm, in particular) may be interested in data before the ErrorFlag cut is applied. This means we must somehow figure out how to apply this selectively. Inside the data handlers we only have access to channels, not to the overall ErrorFlag, so this must likely be done at the data handler array level with a flag :-|

Also, there is a single data point at 5 uA that squeezes through the ErrorFlag cut around pattern 6k.

Finally, the RMS width is overestimated due to the outliers around pattern 17k which are not cut by Device_Error_Code cuts nor ErrorFlag cuts. Not sure what's going on there, but it messes up the correlations due to the large lever arm. What is causing these?

**Comment by wdconinc** [comment]
tl;dr, the QwCorrelator currently determines correlations using the data show in the left panels. It should use the data on the right panels.

**Comment by wdconinc** [comment]
Another way to check this is:
1. Realize that in the first 100k events, 25k patterns (-30 for seed), there is a beam trip that affects 3k patterns so only 22k patterns max should make it into the correlation analysis.
2. Yet, qwparity ends with the line `LinRegBevPeb::printSummaryY seen good eve=24905  (CSV-format)` which is too high and reflects that the beam trip isn't cut out.

**Comment by wdconinc** [comment]
With branch in #208 we get `LinRegBevPeb::printSummaryY seen good eve=21573  (CSV-format)` which is exactly what we get in
```
root [12] mul->Draw("asym_sam1.hw_sum:Entry$","asym_sam1.Device_Error_Code==0 && asym_sam2.Device_Error_Code==0 && asym_sam3.Device_Error_Code==0 && asym_sam4.Device_Error_Code==0 && asym_sam5.Device_Error_Code==0 && asym_sam6.Device_Error_Code==0 && asym_sam8.Device_Error_Code==0 && diff_bpm4eX.Device_Error_Code==0 && diff_bpm4aX.Device_Error_Code==0 && diff_bpm4eY.Device_Error_Code==0 && diff_bpm4aY.Device_Error_Code==0 && diff_bpm12X.Device_Error_Code==0 && ErrorFlag==0")
(long long) 21573
```

**Comment by wdconinc** [comment]
Outliers around pattern 17k not understood yet, but closing anyway since not indicative of an identifiable bug.
